### ISSUE TITLE
./configure --enable-hcache --with-lmdb doesn't define HAVE_LMDB

### ISSUE THREAD

**Issue by 4nd3r** [neomutt/neomutt#87]
```
$ ./configure --enable-hcache --with-lmdb
--8<--
$ grep HAVE_LMDB config.h
/* #undef HAVE_LMDB */
$ grep HAVE_DB4 config.h
#define HAVE_DB4 1
```

```
$ ./configure --enable-hcache --with-tokyocabinet
--8<--
$ grep HAVE_TC config.h
#define HAVE_TC 1
```

```
$ ./configure --enable-hcache --with-lmdb --without-bdb
--8<--
$ grep HAVE_LMDB config.h
#define HAVE_LMDB 1
```

**Comment by flatcap** [comment]
There's a priority order in `configure` which is inherited from Mutt.
The solution, as you found, is to disable the others.
The Travis config script for NeoMutt is explicit:
```
./configure --enable-hcache --without-tokyocabinet --without-qdbm --without-gdbm --without-bdb --with-lmdb
```

I've labelled the issue a question for now.
If @darnir has time to work his magic on `configure`, great.
If not, I will close this issue soon.

**Comment by darnir** [comment]
This part of the code seems to be complex. It'll take some time to fix it up. Let me see what I can do.

After a quick first pass, I haven't completely understood the logic behind it.

**Comment by darnir** [comment]
These would be easier to debug if at least some of the commits from #86 were merged. Being able to display the config status would help a lot.

**Comment by darnir** [comment]
Should now be fixed after commit 553c7ee54f8def79956bf2eab0f87d9ff06ea55e

**Comment by flatcap** [comment]
yep, just tested `./prepare --enable-hcache --with-lmdb`
